### PR TITLE
added perch_v2 with utils for initial testing

### PR DESCRIPTION
**Pull request by vskode**

### CODE REVIEW

**Comment by vskode** [comment]
I had an error because of wandb and protobuf verison problems, so i commented wandb out. You can test perch_v2.py with the 3 lines I added in the bottom. 

You need tensorflow 2.20 and keras 3.11 to run perch_v2.py.

To explain:
I'm using the scripts from perch_hoplite, necessary to run perch_v2. That way we get a results object after inference with logits and embeddings.  tensorflow 2.20 needs cudnn version 9.3. that can conflict with torch so that's why i included a check to make sure that if you don't have cudnn 9.3 in the environment it uses the cpu version of the model. To make sure tf doens't try to access the GPU, i'm setting os.environ["CUDA_VISIBLE_DEVICES"] = "-1". If you run a torch model in the same session, it might not find the GPU's again, so either remove that item from os.environ after the perch_v2.py run or if only one model is run per session I don't think it should cause problems.

I got an error with the `datasets.load_dataset_builder(hf_path, self.hf_name)` so make sure to set restrict_logit's to false or pass the pretrain_info object.

Hope everything works for you.

**Comment by vskode** [comment]
if you want to test on a gpu, installing these requirements worked for me and did not cause problems with pytorch:

```
nvidia-cublas-cu12==12.5.2.13
nvidia-cuda-cccl-cu12==12.5.39.post1
nvidia-cuda-cupti-cu12==12.5.39
nvidia-cuda-cuxxfilt-cu12==12.5.39
nvidia-cuda-nvcc-cu12==12.5.40
nvidia-cuda-nvrtc-cu12==12.5.40
nvidia-cuda-opencl-cu12==12.5.39
nvidia-cuda-profiler-api-cu12==12.5.39
nvidia-cuda-runtime-cu12==12.5.39
nvidia-cuda-sanitizer-api-cu12==12.5.39
nvidia-cudnn-cu12==9.3.0.75
nvidia-cufft-cu12==11.2.3.18
nvidia-curand-cu12==10.3.6.39
nvidia-cusolver-cu12==11.6.2.40
nvidia-cusparse-cu12==12.4.1.24
nvidia-cusparselt-cu12==0.6.2
nvidia-nccl-cu12==2.21.5
nvidia-npp-cu12==12.3.0.116
nvidia-nvfatbin-cu12==12.5.39
nvidia-nvjitlink-cu12==12.5.40
nvidia-nvjpeg-cu12==12.3.2.38
nvidia-nvml-dev-cu12==12.5.39
nvidia-nvtx-cu12==12.5.39
```
